### PR TITLE
Fix/clean namespacing

### PR DESCRIPTION
**Pull request by gtluszcz**
### The problem
In the readme we advise users to use namespaces in their cache store (`namespace: -> { Apartment::Tenant.current }`). In addition in `scoped_store.rb:26` we compute cache_key with `Apartment::Tenant.current` as part of the key. This results in keys like `tenant-1:tenant-1/spree/...`. As you can see they're double namespaced.

in addition...

Because [`spree_shared/scoped_store_decorator.rb`](https://github.com/spree-contrib/spree_shared/blob/master/app/models/spree_shared/scoped_store_decorator.rb) doesn't get loaded before `initializers/spree.rb` an attempt to set configuration like `Spree::Config.layout = 'lol'` will end up using regular spree `ScopedStore#key_for` method with `ENV['RAILS_CACHE_ID']`. It very often is nil which will result in the value being stored  in db under key with suffix `spree/app_configuration/layout` 

However during the request `spree_shared/scoped_store_decorator.rb` is already loaded and [`ScopedStore#key_for`](https://github.com/spree/spree/blob/master/core/app/models/spree/preferences/scoped_store.rb) uses `Apartment::Tenant.current` instead of `ENV['RAILS_CACHE_ID']`. He will look for the same preference under key with suffix `'<current_tenant>/spree/app_configuration/layout'`. Keys when setting and getting values are obviously different, so spree won't find the desired value and will return the default **That's bad**

last but not least overriding `ENV['RAILS_CACHE_ID']` in [`SpreeShared::TenantInitializer`](https://github.com/spree-contrib/spree_shared/blob/master/lib/spree_shared/tenant_initializer.rb) may be surprising to some developers as they may have provided their own values for this env variable

### Solution 
Rely on cache namespace in cache store (`namespace: -> { Apartment::Tenant.current }`) and do not namespace twice. Do not override `ENV['RAILS_CACHE_ID']`. Still use `ENV['RAILS_CACHE_ID']` when computing store items' keys

### Extras
1. I've added points to installation guide to keep it more concise and straightforward  
2. I've ran rubocop on some files to keep them clean
3. I've also allowed myself to create some convenience methods for `Apartment::Tenant` to allow for setting configs in such way: 
```
require 'spree_shared/tenant_decorator'

Apartment::Tenant.each do |tenant_name| # also each_with_default available
  Spree::Config.auto_capture = true
rescue
  puts "  Failed to set up config for store '#{tenant_name}'"
end
```

List of changes:
* Add tests for namespaces
* Improve readme
* Add 'each' methods for tenant
* Update per store config in readme
* Add require to readme
* Run rubocop

### CODE REVIEW

File: spec/spree_shared_spec.rb [hunk 1]
```diff
@@ -21,6 +21,7 @@
     end
     
     Apartment::Tenant.switch!
+    Spree::Preferences::Store.instance.persistence = true
```
Comment (kshalot) [inline spec/spree_shared_spec.rb:24]: Dowiedziałeś się w końcu dlaczego to tutaj musi być? Bo to jest mega nieintuicyjne imo

**Comment by kshalot** [comment]
poskłoszuj komity

**Comment by kshalot** [comment]
Remember to squash the commits :tennis:

File: spec/spree_shared_spec.rb (hunk 1 above)
Comment (kshalot) [inline spec/spree_shared_spec.rb:26]: Did you find out why this is required? The way it works is super weird

File: spec/spree_shared_spec.rb [hunk 2]
```diff
@@ -35,12 +36,61 @@
...
+    end
+
+    it 'iterates over tenants' do
+      expected_tenants = ['tenant271']
+      Apartment.tenant_names = expected_tenants
+
+      tenants = []
+      tenants = Apartment::Tenant.each do |name|
+        tenants << name
+      end
```
Comment (kshalot) [inline spec/spree_shared_spec.rb:102]: Why not just do:
```ruby
tenants = Apartment::Tenant.map { |name| name } end
```

Comment (kshalot) [reply spec/spree_shared_spec.rb:102]: or in one line

File: spec/spree_shared_spec.rb (hunk 1 above)
Comment (gtluszcz) [reply spec/spree_shared_spec.rb:26]: I didn't nope, I would like to learn more about module including/extending/prepending in general. I feel like we'll need that for our reports gem

File: spec/spree_shared_spec.rb (hunk 2 above)
Comment (gtluszcz) [reply spec/spree_shared_spec.rb:102]: there's no map function on this one :D

File: app/models/spree_shared/tenant_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    def default
+        Apartment::Tenant.switch do
+            yield(Apartment::Tenant.current) if block_given?
+        end
+    rescue Apartment::TenantNotFound
+        puts "Failed to connect to default tenant"
+    end
+end
+
+Apartment::Tenant.extend(SpreeShared::TenantDecorator)
```
Comment (grzegorz-wcislo) [inline app/models/spree_shared/tenant_decorator.rb]: Why do you indent by 4 spaces and not 2? Also, you are missing a newline here

**Pull request merged**